integrations/influxdb_connector/influxdb-timestream-connector/template.yml

```diff
@@ -0,0 +1,532 @@
+  LambdaTimeoutInSeconds:
+    Type: Number
+    MinValue: 3
+    Default: 30
+    Description: The amount of time in seconds to run the connector on AWS Lambda before timing out.
+  WriteThrottlingBurstLimit:
+    Type: Number
+    Default: 1200
+    Description: The number of burst requests per second that the REST API Gateway permits.
+
+Conditions:
+  IsAsyncEnabled: !Equals [ !Ref EnableAsyncInvocation, "true" ]
+  IsSyncEnabled: !Equals [ !Ref EnableAsyncInvocation, "false" ]
+
+Resources:
+  RestApiGateway:
+    Type: AWS::ApiGateway::RestApi
+    Properties:
+      Name: !Ref RestApiGatewayName
+
+  ApiResource:
+    Type: AWS::ApiGateway::Resource
+    Properties:
+      ParentId: !GetAtt RestApiGateway.RootResourceId
+      PathPart: api
+      RestApiId: !Ref RestApiGateway
+
+  V2Resource:
+    Type: AWS::ApiGateway::Resource
+    Properties:
+      ParentId: !Ref ApiResource
+      PathPart: v2
+      RestApiId: !Ref RestApiGateway
+
+  WriteResource:
+    Type: AWS::ApiGateway::Resource
+    Properties:
+      ParentId: !Ref V2Resource
+      PathPart: write
+      RestApiId: !Ref RestApiGateway
+
+  SyncRestApiGatewayPostMethod:
+    Type: AWS::ApiGateway::Method
+    Condition: IsSyncEnabled
+    Properties:
+      ResourceId: !Ref WriteResource
+      RestApiId: !Ref RestApiGateway
+      AuthorizationType: AWS_IAM
+      HttpMethod: POST
+      ApiKeyRequired: false
+      Integration:
+        IntegrationHttpMethod: POST
+        Type: AWS_PROXY
+        Uri: !Sub
+          - arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${LambdaArn}/invocations
+          - LambdaArn: !GetAtt LambdaFunction.Arn
+        IntegrationResponses:
+          - StatusCode: '200'
+            SelectionPattern: '2\d{2}'
+            ResponseTemplates:
+              application/json: |
+                {
+                  "statusCode": 200,
+                  "body": "$input.body"
+                }
+          - StatusCode: '500'
+            SelectionPattern: '5\d{2}'
+            ResponseTemplates:
+              application/json: |
+                {
+                  "statusCode": "Internal error",
+                  "error": "$util.escapeJavaScript($input.path('$.errorMessage'))"
+                }
+      MethodResponses:
+        - StatusCode: '200' # Synchronous response, success
+        - StatusCode: '500'
+
+  AsyncRestApiGatewayPostMethod:
+    Type: AWS::ApiGateway::Method
+    Condition: IsAsyncEnabled
+    Properties:
+      ResourceId: !Ref WriteResource
+      RestApiId: !Ref RestApiGateway
+      AuthorizationType: AWS_IAM
+      HttpMethod: POST
+      ApiKeyRequired: false
+      Integration:
+        IntegrationHttpMethod: POST
+        Type: AWS
+        PassthroughBehavior: WHEN_NO_MATCH
+        RequestTemplates:
+          text/plain: |
+            {
+              "path": "$context.path",
+              "httpMethod": "$context.httpMethod",
+              "body" : "$util.escapeJavaScript($input.body)",
```

Comment:
These mapping templates discard the headers, as they serve no purpose and may contain InfluxDB tokens. The most important fields are `body` and `queryStringParameters`. The other fields are included for debugging purposes.